Source/WebCore/html/ImageBitmap.cpp

```diff
@@ -887,7 +887,7 @@ void ImageBitmap::createCompletionHandler(ScriptExecutionContext& scriptExecutio
     // If no cropping, resizing, flipping, etc. are needed, then simply use the
     // resulting ImageBuffer directly.
     auto alphaPremultiplication = alphaPremultiplicationForPremultiplyAlpha(options.premultiplyAlpha);
-    if (sourceRectangle.returnValue().location().isZero() && sourceRectangle.returnValue().size() == imageData->size() && sourceRectangle.returnValue().size() == outputSize && options.orientation == ImageBitmapOptions::Orientation::None) {
+    if (sourceRectangle.returnValue().location().isZero() && sourceRectangle.returnValue().size() == imageData->size() && sourceRectangle.returnValue().size() == outputSize && options.orientation != ImageBitmapOptions::Orientation::FlipY) {
```

Comment:
Would `options.orientation == ImageBitmapOptions::Orientation::FromImage` be nicer here?

Reply:
Honestly, I think this code might just be wrong.

'None' previously meant 'use the image's orientation data', but this fast path is for when there's no rotation to be done (what you would expect 'None' to mean).

I think this should be checking `options.resolvedImageOrientation == None`.

Reply:
I take it back, this is the `ImageData` path, so it can't have its own orientation data.

Writing it as 'if we don't need to flip, then take the fast path' is nicer to me :)

Reply:
ok sounds good!